src/wolff_producer.erl

```diff
@@ -58,12 +59,12 @@
                     max_linger_ms => non_neg_integer(),
                     max_send_ahead => non_neg_integer(),
                     compression => kpro:compress_option(),
-                    drop_if_highmem => boolean()
+                    drop_if_highmem => boolean(),
+                    telemetry_meta_data => term()
                    }.
 
 -define(no_timer, no_timer).
 -define(reconnect, reconnect).
--define(sent_req(Req, Q_AckRef, Calls), {Req, Q_AckRef, Calls}).
```

Comment:
Changing the format of those inflight requests might break a hot-upgrade, unless they are migrated and re-enqueued during `code_upgrade` or the legacy format is handled during the retry, I think.

Reply:
I discussed this with @zmstone. According to him, we can break hot upgradeability and bump the version number. @zmstone shall I bump to version 1.7.0 and then reset wolff.appup.src to:

```
{"1.7.0",
  [],
  []
}.
```

?

Reply:
yes, this new version is going to be used in EMQX 5 but not 4.

Reply:
it happended to be the case here.
however bumping minor or not has little to do with relup.
for pulsar client, we had to support hot-upgrade because we'll try to deliver it in a EMQX patch release.